### DISCUSSION TITLE
Copilot deployment giving error in ECS deployment

### DISCUSSION THREAD

**Discussion by gauravjeswani** [aws/copilot-cli#3067] [Q&A] [answered]
I deployed a copilot app few days back. Which includes a load balancer container and 3 service container. Now I am again trying to deploy same application using copilot, but this time it's giving error in ECS container deployment. I don't understand what's the reason.

Can someone please help me with this?

```
Proposing infrastructure changes for stack test-copilot-loc-test-test-copilot-loc-lb
- Creating the infrastructure for stack test-copilot-loc-test-test-copilot-loc-lb    [rollback complete]  [1320.8s]
  The following resource(s) failed to create: [Service]. Rollback reques
  ted by user.
  - Service discovery for your services to communicate within the VPC                [delete complete]    [0.0s]
  - Update your environment's shared resources                                       [update complete]    [242.9s]
    - A security group for your load balancer allowing HTTP and HTTPS traffic        [create complete]    [5.2s]
    - An Application Load Balancer to distribute public traffic to your services     [create complete]    [183.6s]
  - An IAM Role for the Fargate agent to make AWS API calls on your behalf           [delete complete]    [2.7s]
  - A CloudWatch log group to hold your service logs                                 [delete complete]    [2.7s]
  - An ECS service to run and maintain your tasks in the environment cluster         [delete complete]    [98.5s]
    Resource handler returned message: "Error occurred during operation 'E
    CS Deployment Circuit Breaker was triggered'." (RequestToken: ***********, HandlerErrorCode: GeneralServiceExceptio
    n)
    Deployments
               Revision  Rollout   Desired  Running  Failed  Pending
      PRIMARY  1         [failed]  1        0        10      1
  - A target group to connect the load balancer to your service                      [delete complete]    [0.0s]
  - An ECS task definition to group your containers and run them on ECS              [delete complete]    [0.0s]
  - An IAM role to control permissions for the containers in your tasks              [delete complete]    [2.7s]
X deploy service: stack test-copilot-loc-test-test-copilot-loc-lb did not complete successfully and exited with status ROLLBACK_COMPLETE
```

**Comment by efekarakus** [comment]
Hi @gauravjeswani !

That's very interesting, I haven't seen a `GeneralServiceException` before 🤔 

Would it be possible to go to the ECS console, navigate to your STOPPED tasks to see the reason? There might be more details there:
![stopped](https://user-images.githubusercontent.com/879348/142916976-b53583a5-9803-4a0a-adb4-fae17cd1ad26.png)

**Comment by gauravjeswani** [comment, marked as answer]
Hello @efekarakus ,

Thanks for your reply. I found the rootcause of issue. Basically previously I deployed Backed Service first and then followed by Load Balance. And this time somehow I tried for LoadBalance first. Which was causing the issue.

Because LoadBalancer was Nginx container with .conf file pointing to service containers. As service URLs were not ready yet as I was deploying LoadBalancer first. Container was breaking and was not able to start.

I think it would be great if we can show correct error in copilot rather than this weird error.
